### PR TITLE
tbarri dev/auto publish

### PR DESCRIPTION
**Pull request by tbarri**
- Sequentially publish NPM dependencies and remove workspaces during publish
- Simplify NPM dependencies between the IDL, Core and CLI packages
- Remove dead testing boilerplate code from the CLI
- Default linting rules to that specified in the root package.json file
- Update CLI README with auto-generated oclif content
- Update YARN.lock dependencies

### CODE REVIEW

File: solana/program_v2/packages/client/cli/src/util/util.ts [hunk 1]
```diff
@@ -36,16 +36,16 @@ export const parseNetworkUpdateData = (
...
-  });
+  }
   return updateData;
 };
 
 export const parseGatekeeperUpdateData = (
   updateData: UpdateGatekeeperData
 ): UpdateGatekeeperData => {
   const keysArr = Object.keys(updateData);
-  // eslint-disable-next-line unicorn/no-array-for-each
```
Comment (JulianSpring) [inline solana/program_v2/packages/client/cli/src/util/util.ts:47]: Weird thing to disable?

**Review by JulianSpring — APPROVED** [review]
Really nice clean up. Thanks!

File: .github/workflows/npm-auto-publish-workflow.yml [hunk 2]
```diff
@@ -54,6 +59,12 @@ jobs:
             echo "tag-value=beta" >> $GITHUB_OUTPUT
           fi
 
+      - name: Remove workspace configuration
+        working-directory: ${{ inputs.root-directory }}
+        run: |
+          jq 'del(.workspaces)' package.json >> package-edited.json
```
Comment (JulianSpring) [inline .github/workflows/npm-auto-publish-workflow.yml:65]: Should we point this out? That is is kind of a hack?

Comment (JulianSpring) [reply .github/workflows/npm-auto-publish-workflow.yml:75]: Awesome thank you! :)

**Review by JulianSpring — APPROVED** [review]

**Review by william-brooks — APPROVED** [review]

File: solana/program_v2/packages/tests/package.json [hunk 3]
```diff
@@ -3,18 +3,19 @@
...
-    "lint": "eslint src/**/*.ts"
+    "build": "tsc --build"
   },
   "dependencies": {
-    "@identity.com/gateway-solana-client": "^2.0.0-beta.8",
-    "@identity.com/gateway-solana-idl": "^2.0.0-beta.8",
-    "@project-serum/anchor": "^0.26.0",
-    "@solana/spl-token": "^0.3.5",
-    "@solana/web3.js": "^1.41.3"
+    "@identity.com/gateway-solana-client": "^2.0.0-beta.12",
```
Comment (rado0x54) [inline solana/program_v2/packages/tests/package.json:9]: I think technically everything in the `test` project should be a `devDependency` :)

File: solana/program_v2/packages/tests/package.json [hunk 4]
```diff
@@ -3,18 +3,19 @@
...
-    "build": "tsc --build",
-    "lint": "eslint src/**/*.ts"
+    "build": "tsc --build"
   },
   "dependencies": {
-    "@identity.com/gateway-solana-client": "^2.0.0-beta.8",
-    "@identity.com/gateway-solana-idl": "^2.0.0-beta.8",
-    "@project-serum/anchor": "^0.26.0",
-    "@solana/spl-token": "^0.3.5",
-    "@solana/web3.js": "^1.41.3"
```
Comment (rado0x54) [inline solana/program_v2/packages/tests/package.json:14]: Also used in `utils.ts`.

File: solana/program_v2/packages/client/core/package.json [hunk 5]
```diff
@@ -21,16 +20,16 @@
...
-    "build": "tsc -p tsconfig.json"
+    "build": "tsc -p tsconfig.json",
+    "prepack": "yarn install && yarn build"
   },
   "dependencies": {
-    "@identity.com/gateway-solana-idl": "^2.0.0-beta.8",
+    "@identity.com/gateway-solana-idl": "^2.0.0-beta.12",
+    "@solana/web3.js": "^1.50.1",
     "@project-serum/anchor": "^0.26.0",
     "@project-serum/anchor-cli": "^0.26.0"
```
Comment (rado0x54) [inline solana/program_v2/packages/client/core/package.json:30]: I actually don't see why that needs a `anchor-cli` and if it does it should only be a `devDependency`. please remove

Comment (rado0x54) [reply solana/program_v2/packages/client/core/package.json:30]: (I understand that this was not changed in the PR, but needs to be corrected. :))

File: solana/program_v2/packages/client/cli/package.json [hunk 6]
```diff
@@ -13,35 +13,16 @@
...
-    "eslint-config-oclif": "^4",
-    "eslint-config-oclif-typescript": "^1.0.2",
-    "globby": "^11",
-    "mocha": "^9",
     "oclif": "^3",
-    "shx": "^0.3.3",
-    "ts-mocha": "^10.0.0",
-    "ts-node": "^10.2.1",
-    "tslib": "^2.3.1",
+    "@oclif/core": "^1",
```
Comment (rado0x54) [inline solana/program_v2/packages/client/cli/package.json:23]: That's directly used in `add-gatekeeper.ts` and therefore I think it should be a `dependency`. (But double check with the oclif framework)

File: solana/program_v2/packages/client/cli/package.json [hunk 7]
```diff
@@ -13,35 +13,16 @@
...
     "/oclif.manifest.json"
   ],
   "dependencies": {
-    "@identity.com/gateway-solana-client": "^2.0.0-beta.8",
-    "@project-serum/anchor": "^0.26.0",
-    "@project-serum/anchor-cli": "^0.26.0",
-    "@oclif/core": "^1",
-    "@oclif/plugin-help": "^5",
-    "@oclif/plugin-plugins": "^2.0.1",
-    "@solana/web3.js": "^1.50.1"
```
Comment (rado0x54) [inline solana/program_v2/packages/client/cli/package.json:26]: Also used directly. `@solana/web3.js`

**Review by rado0x54 — APPROVED** [review]

**Pull request merged**